pages/common/helpers/index.js

```diff
@@ -16,8 +16,16 @@ const getNacwoById = (req, id) =>
     .then(nacwo => nacwo && nacwo.profile)
     .catch(err => Promise.reject(err));
 
+const saveTaskIdToSession = session => response => {
+  session.success = {
+    taskId: get(response, 'json.data.id')
+  };
+  return response;
+};
```

Comment:
I know _why_ you have done this, but it seems a little too side-effecty for a helper function, it relies on being called in a very specific context, I think it would feel nicer to have an implementation that looks more like this

```
.then(response => {
  saveTaskIdToSession(req, response);
  return response;
})
```

but then it probably doesn't warrant having a helper at all, maybe I need to care less

Reply:
originally there was no helper function and every router was saving to the session itself - but yeah if I simplify this any further (e.g. passing in the taskId instead of the response) it feels like I might as well just go back to not having a helper.

Reply:
I think the part that bothers me is the need to return the response, perhaps it would be better to have this as the last step in the promise chain, before catch, then it can just do it's thing without caring about what comes next?

Reply:
also "maybe I need to care less"

Reply:
I realise that the "calling last" thing is not an option, because redirects, so please disregard that 🤷